native_client/deepspeech.cc

```diff
@@ -8,45 +8,423 @@
 #endif
 
 #include <iostream>
+#include <vector>
 
 #include "deepspeech.h"
-#include "deepspeech_utils.h"
 #include "alphabet.h"
 #include "beam_search.h"
 
 #include "tensorflow/core/public/session.h"
 #include "tensorflow/core/platform/env.h"
 #include "tensorflow/core/util/memmapped_file_system.h"
 
-#define BATCH_SIZE 1
+#include "c_speech_features.h"
+
+//TODO: infer batch size from model/use dynamic batch size
+const int BATCH_SIZE = 1;
+
+//TODO: use dynamic sample rate
+const int SAMPLE_RATE = 16000;
+
+//TODO: infer n_steps from model
+const int N_STEPS_PER_BATCH = 16;
```

Comment:
Those three `TODO` about using data from the model, seems related to my previous comment about relying on some graph metadata somehow.

Reply:
Yeah, n_steps and batch_size should definitely be obtained from the model directly. I think they're both easy to do, FWIW, just haven't gotten to it yet.